### PR TITLE
Add resource_type on keycloak_openid_client_authorization_permission

### PR DESCRIPTION
**Pull request by JessieAMorris**
See issue mrparkers/terraform-provider-keycloak#398 for details. Also
adds docs for keycloak_openid_client_authorization_permission as well.

### CODE REVIEW

**Comment by JessieAMorris** [comment]
@mrparkers any chance you could review this? This is blocking me from implementing some features and the changes are pretty minimal.

**Comment by JessieAMorris** [comment]
@mrparkers any word on this? I don't want this MR to languish and fall by the wayside. I see other MRs getting merged so some feedback on what I need to change or do to get this merged would be great.

**Comment by mrparkers** [comment]
I'm sorry @JessieAMorris, you're right that I've been neglecting this one.  At a glance the code LGTM, and I also really appreciate adding the docs for this resource too.  Give me an hour or two to pull this down locally and get a feel for it, and I'll get this merged and cut a release for you today.

Thanks for your patience!

File: provider/resource_keycloak_openid_client_authorization_permission.go [hunk 1]
```diff
@@ -59,6 +60,10 @@ func resourceKeycloakOpenidClientAuthorizationPermission() *schema.Resource {
 				Elem:     &schema.Schema{Type: schema.TypeString},
 				Optional: true,
 			},
+			"resource_type": {
+				Type:     schema.TypeString,
+				Optional: true,
```
Comment (mrparkers) [inline provider/resource_keycloak_openid_client_authorization_permission.go:65]: In the GUI, it looks like you can't set the resource type and a list of resources.  Should we add a `ConflictsWith` here and within the `resources` attribute?

Comment (mrparkers) [reply provider/resource_keycloak_openid_client_authorization_permission.go]: I went ahead and pushed this change to your branch so I could get this merged / released today.  If it turns out you don't think this is needed, we can remove it in a follow-up PR.

**Review by mrparkers — APPROVED** [review]
Thanks again for the PR, so sorry for the super long delay on this one.

File: provider/resource_keycloak_openid_client_authorization_permission.go (hunk 1 above)
Comment (JessieAMorris) [reply provider/resource_keycloak_openid_client_authorization_permission.go]: Ah, thanks. I had that change being made but was running acceptance tests! Sorry about the delay getting back to you, it's been a crazy morning.

**Comment by JessieAMorris** [comment]
Thanks for the merge! Looking forward to using this in my code. :-)

File: provider/resource_keycloak_openid_client_authorization_permission.go (hunk 1 above)
Comment (mrparkers) [reply provider/resource_keycloak_openid_client_authorization_permission.go]: You have nothing to apologize for, I'm pretty sure my delay in reviewing this PR was longer 😉

**Pull request merged**